content/en/blog/_posts/2021-08-09-memory-manager-moves-to-beta.md

```diff
@@ -0,0 +1,108 @@
+2. The **MemoryManager** get the NUMA nodes where it allocated the memory for the container and returns it to **Kubelet**.
+3. The **Kubelet** creates the container via CRI with the spec updated according to the **MemoryManager** information.
+
+### Let's talk about the configuration
+
+By default, the **MemoryManager** runs with the `None` policy, meaning it will just relax and will not do anything. To enable the **MemoryManager**, you should provide two **Kubelet** flags:
+
+- `--memory-manager-policy=Static`
+- `--reserved-memory="<numaNodeID>:<resourceName>=<quantity>"`
+
+It is clear what value you should set for the `--memory-manager-policy` parameter. But it is not 100% clear how you should configure the `--reserved-memory` parameter. To configure it correctly, you should follow two main rules:
+
+- The amount of reserved memory for the `memory` resource must be greater than zero.
+- The amount of reserved memory for the resource type must be equal to [NodeAllocatable](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/) (`kube-reserved + system-reserved + eviction-hard`) for the resource. You can read more about
+
+![ReservedMemory
+](/images/blog/2021-08-09-memory-manager-moves-to-beta/ReservedMemory.svg "ReservedMemory")
+
+## Current limitations
+
+The 1.22 release and promotion to beta brings along enhancements and fixes, but the **MemoryManager** still has several limitations.
+
+### Single vs Cross NUMA node allocation
+
+The NUMA node can not have both single and cross NUMA node allocations. When the container memory is pinned to two or more NUMA nodes, we can not know from which NUMA node the container will consume the memory.
+
+![SingleCrossNUMAAllocation
+](/images/blog/2021-08-09-memory-manager-moves-to-beta/SingleCrossNUMAAllocation.svg "SingleCrossNUMAAllocation")
+
+1. The `container1` started on the NUMA node 0 and requests *5Gi* of the memory but currently is consuming only *3Gi* of the memory.
+2. For container2 the memory request is 10Gi, and no single NUMA node can satisfy it.
+3. The `container2` consumes *3.5Gi* of the memory from the NUMA node 0, but once the `container1` will require more memory, it will not have it, and the kernel will kill one of the containers with the **OOM** error.
+
+To prevent such issues, the **MemoryManager** will fail the admission of the `container2` until the machine has two NUMA nodes without a single NUMA node allocation.
+
+### Works only for guaranteed pods
+
+The **MemoryManager** can not guarantee memory allocation for burstable pods, also when the burstable pod has specified equal memory limit and request.
+Let's assume you have two burstable pods, `pod1` has containers with equal memory request and limits, and `pod2` has containers only with the memory request. We want to guarantee memory allocation for the `pod1`. Both pods have the same **OOM** score, so once the kernel will not have enough memory, it can kill `pod1`.
```

Comment:
I'd write:
```suggestion
Let's assume you have two Burstable pods: `pod1` has containers with
equal memory request and limits, and `pod2` has containers only with a
memory request set. You want to guarantee memory allocation for the `pod1`.
To the Linux kernel, processes in either pod have the same *OOM score*, 
o once the kernel finds that it does not have enough memory, it can kill 
processes that belong to pod `pod1`.
```